File: frontend/graylink-web/src/types/file.ts
export interface FileItem {
  name: string
  path: string
  type: 'file' | 'directory'
  size?: number
  modified?: string
  extension?: string
}

export type FileOperations = 'copy' | 'move' | 'delete'

export interface FileState {
  currentPath: string
  files: FileItem[]
  selectedFiles: string[]
  sortBy: string
  sortDesc: boolean
  loading: boolean
}

export interface FileInfo extends FileItem {
  id: string
  parent_id?: string
  created_at: string
  updated_at: string
}

export interface BatchOperationParams {
  operation: FileOperations
  paths: string[]
  targetPath?: string
}

export interface FileApiResponse {
  items: FileItem[]
  total: number
  path: string
}

File: backend/schemas/file.py
from pydantic import BaseModel
from typing import List, Optional
from enum import Enum

class FileOperation(str, Enum):
    COPY = 'copy'
    MOVE = 'move'
    DELETE = 'delete'

class BatchOperationRequest(BaseModel):
    operation: FileOperation
    paths: List[str]
    target_path: Optional[str] = None

class FileInfo(BaseModel):
    name: str
    path: str
    type: str
    size: Optional[int] = None
    modified: Optional[str] = None
    extension: Optional[str] = None 